testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -5,6 +5,13 @@
 # Texts and links
 #
 
+def wait_for_spa_transition(text)
+  repeat_until_timeout(timeout: 500, message: "Error on navigating to '#{text}'") do
+    break unless page.execute_script('return window.pageRenderers.spa.isInTransition()')
+    sleep 1
```

Comment:
I don't like the idea to almost always wait one second for each action we do. As at the point isInTransition() is executed most probably still not finished. Reduce it to at least `sleep 0.2`

Reply:
Done. Global timeout is now 10 and sleep 0.2